File: LibLiteLog/src/main/java/io/github/tomgarden/lib/lite_log/Printer.java
package io.github.tomgarden.lib.lite_log;

/**
 * A proxy interface to enable additional operations.
 * <p>
 * 一个允许增加额外操作的代理接口
 * <p>
 * Contains all possible Log message usages.
 * <p>
 * 包含所有的日志用法
 */
public abstract class Printer {

    /*默认控制台日志策略*/
    private LogStrategy defLogcatStrategy = null;

    /*临时控制台日志策略*/
    private LogStrategy temporaryLogcatStrategy = null;

    /*默认磁盘日志策略*/
    private LogStrategy defDiskStrategy = null;

    /*临时磁盘日志策略*/
    private LogStrategy temporaryDiskStrategy = null;



    public LogStrategy getDefLogcatStrategy() {
        return defLogcatStrategy;
    }

    public void setDefLogcatStrategy(LogStrategy defLogcatStrategy) {
        this.defLogcatStrategy = defLogcatStrategy;
    }

    public LogStrategy getTemporaryLogcatStrategy() {
        return temporaryLogcatStrategy;
    }

    public void setTemporaryLogcatStrategy(LogStrategy temporaryLogcatStrategy) {
        this.temporaryLogcatStrategy = temporaryLogcatStrategy;
    }

    public LogStrategy getDefDiskStrategy() {
        return defDiskStrategy;
    }

    public void setDefDiskStrategy(LogStrategy defDiskStrategy) {
        this.defDiskStrategy = defDiskStrategy;
    }

    public LogStrategy getTemporaryDiskStrategy() {
        return temporaryDiskStrategy;
    }

    public void setTemporaryDiskStrategy(LogStrategy temporaryDiskStrategy) {
        this.temporaryDiskStrategy = temporaryDiskStrategy;
    }

    abstract LogStrategy unNullTemporaryLogcatStrategy();

    abstract public LogStrategy unNullTemporaryDiskStrategy();

    abstract void d(String message, Object... args);

    abstract void d(Object any);

    abstract void e(Boolean withSingleFile, String message, Object... args);

    abstract void e(Throwable throwable, Boolean withSingleFile, String message, Object... args);

    abstract void w(String message, Object... args);

    abstract void i(String message, Object... args);

    abstract void v(String message, Object... args);

    abstract void wtf(String message, Object... args);

    abstract void log(int priority, String message, Throwable throwable, boolean withSingleFile);

    abstract void clearLogStrategies();
}

File: LibLog/src/main/java/io/github/tomgarden/lib/log/Printer.kt
package io.github.tomgarden.lib.log

/**
 * A proxy interface to enable additional operations.
 *
 * 一个允许增加额外操作的代理接口
 *
 * Contains all possible Log message usages.
 *
 * 包含所有的日志用法
 */
interface Printer {

  /*默认控制台日志策略*/
  var defLogcatStrategy: LogStrategy?
  /*临时控制台日志策略*/
  var temporaryLogcatStrategy: LogStrategy?
  /*默认磁盘日志策略*/
  var defDiskStrategy: LogStrategy?
  /*临时磁盘日志策略*/
  var temporaryDiskStrategy: LogStrategy?

  fun unNullTemporaryLogcatStrategy(): LogStrategy

  fun unNullTemporaryDiskStrategy(): LogStrategy

  fun d(message: String?, vararg args: Any)

  fun d(any: Any?)

  fun e(withSingleFile: Boolean, message: String?, vararg args: Any)

  fun e(throwable: Throwable?, withSingleFile: Boolean, message: String?, vararg args: Any)

  fun w(message: String?, vararg args: Any)

  fun i(message: String?, vararg args: Any)

  fun v(message: String?, vararg args: Any)

  fun wtf(message: String?, vararg args: Any)

  /**
   * Formats the given json content and print it
   */
  fun json(json: String?)

  /**
   * Formats the given xml content and print it
   */
  fun xml(xml: String?)

  fun log(priority: Int, message: String?, throwable: Throwable?, withSingleFile: Boolean)

  fun clearLogStrategies()
}
